lib/lib-storage/src/Upload.ts

```diff
@@ -66,6 +66,7 @@ export class Upload extends EventEmitter {
 
   private uploadedParts: CompletedPart[] = [];
   private uploadEnqueuedPartsCount = 0;
+  private expectedPartsCount = 0;
```

Comment:
it should not be assigned a value here, since 0 is never possible and a value is always assigned in the constructor.

But since my comments below understand this value could be impossible to calculate, in which case it should be `undefined`. Therefore the type should be `private expectedPartsCount?: number;`.